ui/components/app/snaps/snap-ui-renderer/components/address-input.test.ts

```diff
@@ -0,0 +1,153 @@
+import { Box, AddressInput, Field } from '@metamask/snaps-sdk/jsx';
+import { fireEvent } from '@testing-library/react';
+import { renderInterface } from '../test-utils';
+
+describe('SnapUIAddressInput', () => {
+  it('supports address inputs', () => {
+    const { container, getByRole } = renderInterface(
+      Box({
+        children: AddressInput({
+          name: 'input',
+          chainId: 'eip155:0',
+        }),
+      }),
+    );
+
+    const input = getByRole('textbox') as HTMLInputElement;
+    expect(input).toBeDefined();
+    expect(input.value).toStrictEqual('');
+
+    expect(container).toMatchSnapshot();
+  });
+
+  it('supports address inputs with existing state', () => {
+    const { container, getByRole } = renderInterface(
+      Box({
+        children: AddressInput({
+          name: 'input',
+          chainId: 'eip155:0',
+        }),
+      }),
+      {
+        state: { input: 'eip155:0:0x1234567890123456789012345678901234567890' },
+      },
+    );
+
+    const input = getByRole('textbox') as HTMLInputElement;
+    expect(input).toBeDefined();
+    expect(input.value).toStrictEqual(
+      '0x1234567890123456789012345678901234567890',
+    );
+
+    expect(container).toMatchSnapshot();
+  });
+
+  it('supports address inputs with a placeholder', () => {
+    const { container, getByRole } = renderInterface(
+      Box({
+        children: AddressInput({
+          name: 'input',
+          placeholder: 'Enter an address',
+          chainId: 'eip155:0',
+        }),
+      }),
+    );
+
+    const input = getByRole('textbox') as HTMLInputElement;
+    expect(input).toBeDefined();
+    expect(input.placeholder).toStrictEqual('Enter an address');
+
+    expect(container).toMatchSnapshot();
+  });
+
+  it('supports address inputs with a disabled prop', () => {
+    const { container, getByRole } = renderInterface(
+      Box({
+        children: AddressInput({
+          name: 'input',
+          disabled: true,
+          chainId: 'eip155:0',
+        }),
+      }),
+    );
+
+    const input = getByRole('textbox') as HTMLInputElement;
+    expect(input).toBeDefined();
+    expect(input).toBeDisabled();
+
+    expect(container).toMatchSnapshot();
+  });
+
+  it('supports address inputs within a field', () => {
+    const { container, getByRole, getByText } = renderInterface(
+      Box({
+        children: [
+          Field({
+            label: 'Address',
+            children: AddressInput({
+              name: 'input',
+              chainId: 'eip155:0',
+            }),
+          }),
+        ],
+      }),
+    );
+
+    const input = getByRole('textbox') as HTMLInputElement;
+    expect(input).toBeDefined();
+    const label = getByText('Address');
+    expect(label).toBeDefined();
+
+    expect(container).toMatchSnapshot();
```

Comment:
Do we need a snapshot for each test in this file?

Reply:
Removed a few.